client/src/components/ProfileEdit.jsx

```diff
@@ -0,0 +1,184 @@
+import React, { Component } from 'react';
+import { Row, Col, Input, Button } from 'react-materialize'
+import { request } from '../utils'
+import { Link } from 'react-router-dom'
+
+
+class ProfileEdit extends Component {
+
+    state = {
+        user_edit: {},
+        error_nick: "",
+        error_repeated: "",
+        error_first: "",
+        error_last: "",
+        error_trigger_nick: false,
+        error_trigger_first: false,
+        error_trigger_last: false
+    }
+
+    componentDidMount() {
+        request('/profile/' + this.props.match.params.profile_id)
+        .then(json => {
+            this.setState({user_edit: json});
+            this.setState({
+                user_edit: {
+                    f_name: this.state.user_edit.first_name,
+                    l_name: this.state.user_edit.last_name,
+                    nick: this.state.user_edit.nickname,
+                    loc: this.state.user_edit.location,
+                    pass: this.state.user_edit.password
+                }
+            })
+        })
+    }
+
+    UpdateClick = (eve) => {
+        eve.preventDefault()
+
+        let msg_for_repeated = ""
+
+        this.setState({ error_trigger_nick: false })
+        this.setState({ error_trigger_first: false })
+        this.setState({ error_trigger_last: false })
+
+        this.setState({ error_repeated: ""})
+
+        if (this.state.new_password === this.state.repeat_password) {
+            const data = {
+                "first_name": this.state.f_name,
+                "last_name": this.state.l_name,
+                "nickname": this.state.nickname,
+                "location": this.state.location,
+                "password": this.state.new_password,
+                "id": this.state.user_edit.id
+            }
+            request('/profile/' + this.props.match.params.profile_id, "PUT", JSON.stringify(data))
+            .then(data => {
+                if (data.check_nick === true) {
```

Comment:
It's a bit complicated. You may receive a list of error fields from server along with error messages and apply it form dynamically without hardcoding.